httptransport/httptransport.go

```diff
@@ -0,0 +1,201 @@
+package httptransport
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+
+	clairerror "github.com/quay/clair/v4/clair-error"
+	"github.com/quay/clair/v4/config"
+	"github.com/quay/clair/v4/indexer"
+	"github.com/quay/clair/v4/matcher"
+	"github.com/quay/clair/v4/middleware/auth"
+	intromw "github.com/quay/clair/v4/middleware/introspection"
+	"github.com/rs/zerolog"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/plugin/othttp"
+)
+
+const (
+	apiRoot                 = "/api/v1/"
+	VulnerabilityReportPath = apiRoot + "vulnerability_report/"
+	IndexAPIPath            = apiRoot + "index_report"
+	IndexReportAPIPath      = apiRoot + "index_report/"
+	StateAPIPath            = apiRoot + "state"
+)
+
+// HttpTransport is the primary http server
+// Clair exposes it's functionality on.
+type HttpTransport struct {
```

Comment:
`Server` would be a better name for this. `httptransport.HttpTransport` vs `httptransport.Server`

Reply:
@hdonnay  cool, I'm doing the same for introspection package.
httptransport.Server
introspection.Server